File: src/Exporter.NX.Entry/ExportJob.cs
// SPDX-License-Identifier: GPL-2.0-only
using Oblikovati.Exporter.NX.Nx;
using Oblikovati.Exporter.NX.Recipe;
using Oblikovati.Exporter.NX.Translate;

namespace Oblikovati.Exporter.NX.Entry
{
    /// <summary>
    /// The whole export, end to end and free of NXOpen: read the work document, translate
    /// the tree, write every file to <paramref name="sink"/>, and return the user summary.
    /// The NX entry point supplies a live session and a directory sink; tests supply fakes.
    /// </summary>
    public static class ExportJob
    {
        public static string Run(INxSession session, IDocumentSink sink)
        {
            var runner = new ExportRunner(session, new DocumentTranslator(), new RecipeYamlWriter());
            ExportOutput output = runner.Run();
            foreach (ExportFile file in output.Files)
            {
                sink.Write(file.FileName, file.Yaml);
            }

            return ExportReportFormatter.Summarize(output);
        }
    }
}

File: python/oblikovati_exporter_nx/entry/export_job.py
# SPDX-License-Identifier: GPL-2.0-only
"""The whole export, end to end and free of NXOpen.

Read the work document, translate the tree, write every file to the sink, and return the
user summary. The NX journal supplies a live session and a directory sink; tests supply
fakes.
"""
from __future__ import annotations

from ..recipe.yaml_writer import RecipeYamlWriter
from ..translate.document_translator import DocumentTranslator
from .document_sink import DocumentSink
from .export_runner import ExportRunner
from . import report_formatter


def run(session, sink: DocumentSink) -> str:
    runner = ExportRunner(session, DocumentTranslator(), RecipeYamlWriter())
    output = runner.run()
    for file in output.files:
        sink.write(file.file_name, file.yaml_text)
    return report_formatter.summarize(output)
